File: 1065 한수/1065.java
import java.io.IOException;
import java.util.Scanner;

public class Main {
    static public void main(String[] args) throws IOException {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int[] counter = new int[n + 1];

        for (int i = 1; i <= n; ++i) {
            if (i < 100) {
                counter[i] = counter[i - 1] + 1;
            } else {
                String numberStr = String.valueOf(i);
                int numberStrLen = numberStr.length();
                int diff = numberStr.charAt(1) - numberStr.charAt(0);
                boolean isOk = true;
                for (int j = 1; j < numberStrLen; ++j) {
                    if (numberStr.charAt(j) - numberStr.charAt(j - 1) != diff) {
                        isOk = false;
                        break;
                    }
                }

                counter[i] = isOk ? counter[i - 1] + 1 : counter[i - 1];
            }
        }
        System.out.println(counter[n]);
    }
}

File: 1065 한수/1065.py
n = int(input())
counter = [0 for _ in range(0, n + 1)]

for i in range(1, n + 1):
    if i < 100:
        counter[i] = counter[i - 1] + 1
    else:
        numberStr = str(i)
        numberStrLen = len(numberStr)
        diff = ord(numberStr[1]) - ord(numberStr[0])
        isOk = True
        for j in range(1, numberStrLen):
            if ord(numberStr[j]) - ord(numberStr[j - 1]) != diff:
                isOk = False
                break

        counter[i] = counter[i - 1] + 1 if isOk else counter[i - 1]

print(counter[n])
